microsoft/testsuites/dpdk/dpdksuite.py

```diff
@@ -0,0 +1,252 @@
+    _ninja_url = (
+        "https://github.com/ninja-build/ninja/releases/download/v1.10.2/ninja-linux.zip"
+    )
+
+    _ip_addr_regex = r"[0-9]+:\s+([a-zA-Z0-9\-_]+):\s+<(?:[a-zA-Z0-9_]+,?)+>"
+
+    def _install_dpdk_dependencies(self, node: Node) -> None:
+        if isinstance(node.os, Ubuntu):
+            for package in self._ubuntu_packages:
+                node.os.install_packages(package)
+            self.log.info("Packages installed for Ubunutu")
+            self._execute_expect_zero(node, "pip3 install --upgrade meson")
+            self._execute_expect_zero(node, "mv /usr/bin/meson /usr/bin/meson.bak")
+            self._execute_expect_zero(node, "ln -s /usr/local/bin/meson /usr/bin/meson")
+            self._execute_expect_zero(node, "pip3 install --upgrade ninja")
+
+        elif isinstance(node.os, Redhat) or isinstance(node.os, CentOs):
+            self._execute_expect_zero(
+                node, "yum update -y --disablerepo='*' --enablerepo='*microsoft*'"
+            )
+            node.os.install_packages(
+                ["groupinstall", "'Infiniband Support'"], signed=False
+            )  # todo gross hack to support groupinstall
+            for package in self._redhat_packages:
+                node.os.install_packages(package)
+            result = node.execute(
+                "dracut --add-drivers 'mlx4_en mlx4_ib mlx5_ib' -f"
+            )  # add mellanox drivers
+            self.log.debug("\n".join([result.stdout, result.stderr]))
+            self._execute_expect_zero(node, "systemctl enable rdma")
+            self._execute_expect_zero(node, "pip3 install --upgrade meson")
+            self._execute_expect_zero(node, "ln -s /usr/local/bin/meson /usr/bin/meson")
+
+            self._execute_expect_zero(
+                node,
+                f"wget {self._ninja_url}",
+            )
+            self._execute_expect_zero(
+                node, "unzip ninja-linux.zip && mv ninja /usr/bin/ninja"
+            )
+            self._execute_expect_zero(node, "pip3 install --upgrade pyelftools")
+
+    def _execute_expect_zero_with_path(
+        self, node: Node, cmd: str, path: PurePath
+    ) -> str:
+        result = node.execute(cmd, sudo=True, cwd=path, shell=True)
+        assert_that(result.exit_code).described_as(
+            f"{cmd} failed with code {result.exit_code} and stdout+stderr:"
+            + f"\n{result.stdout}\n=============\n{result.stderr}\n=============\n"
+        ).is_zero()
+        self.log.info(f"{cmd}:\n{result.stdout}")  # TODO: debug
+
+        return result.stdout
+
+    def _execute_expect_zero(self, node: Node, cmd: str) -> str:
+        return self._execute_expect_zero_with_path(node, cmd, node.working_path)
+
+    def _hugepages_init(self, node: Node) -> None:
+        self._execute_expect_zero(node, "mkdir -p /mnt/huge")
+        self._execute_expect_zero(node, "mkdir -p /mnt/huge-1G")
+        self._execute_expect_zero(node, "mount -t hugetlbfs nodev /mnt/huge")
```

Comment:
Lili wrote a tool called `mount` already. Please reuse it, and extend it if it's necessary.